logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/message/MessageTarget.kt

```diff
@@ -20,7 +20,19 @@ package com.wire.kalium.logic.feature.message
 
 import com.wire.kalium.logic.data.conversation.Recipient
 
-sealed class MessageTarget {
-    class Client(val recipients: List<Recipient>) : MessageTarget()
-    object Conversation : MessageTarget()
+sealed interface MessageTarget {
+
+    sealed interface Client : MessageTarget {
+        val recipients: List<Recipient>
+
+        data class ReportIfMissing(
+            override val recipients: List<Recipient>
+        ) : Client
+
+        data class IgnoreIfMissing(
+            override val recipients: List<Recipient>
+        ) : Client
+    }
```

Comment:
Since `report_only` is operating on users and not clients doesn't it makes more sense to have something like

```
sealed interface MessageTarget {
  class Client(val recipients: List<Recipient>) : MessageTarget() // Maps to ignoreAll
  class User(val users: List<UserId>) : MessageTarget() // Maps to reportOnly
  object Conversation : MessageTarget // Maps to reporAll
}
```